src/drivers/fluid_adriver.c

```diff
@@ -323,7 +323,20 @@ new_fluid_audio_driver(fluid_settings_t *settings, fluid_synth_t *synth)
 
     if(def)
     {
-        fluid_audio_driver_t *driver = (*def->new)(settings, synth);
+        fluid_audio_driver_t *driver;
+        double srate, latency;
+        int period_size;
+        
+        fluid_settings_getint(settings, "audio.period-size", &period_size);
+        fluid_settings_getnum(settings, "synth.sample-rate", &srate);
+        
+        latency = period_size / srate;
+        if(latency >= 0.05)
+        {
+            FLUID_LOG(FLUID_WARN, "You have chosen 'audio.period-size' to be %d samples. Given a sample rate of %.1f this results in a latency of %.1f ms, which will cause MIDI events to be poorly quantized (=untimed) in the synthesized audio (also known as the 'drunken-drummer' syndrome). To avoid that, you're strongly advised to increase 'audio.periods' instead, while keeping 'audio.period-size' small enough to make this warning disappear.", period_size, srate, latency*1000.0);
```

Comment:
'audio.periods' is ignored by half of the audio drivers: pulseaudio and pipewire are the most relevant among Linux users.

Reply:
The pipewire hasn't even been released yet in fluidsynth. I consider the ALSA driver to be most relevant on Linux, esp. because it's the default. Yet, I welcome your PR to add support to other drivers.

Reply:
PulseAudio PR: #1127